cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -216,6 +271,15 @@ public static CruiseControl fromCrd(Reconciliation reconciliation, Kafka kafkaAs
             cruiseControl.setTlsSidecar(tlsSidecar);
 
             cruiseControl = cruiseControl.updateConfiguration(spec);
+            CruiseControlConfiguration c = (CruiseControlConfiguration) cruiseControl.getConfiguration();
+            Boolean sslEnabled = isApiAuthenticationEnabled(c);
+            if (sslEnabled) {
+                cruiseControl.uriScheme = HTTPS_SCHEME;
+            } else {
+                cruiseControl.uriScheme = HTTP_SCHEME;
+            }
```

Comment:
I'm a bit confused. From some parts it seems like the authentication is based on some username / password. But here you seem to use it only to enable the TLS.

Reply:
Yes, this part only to sets the scheme used by the probes without setting up the certs/keys since the kubelet skips certificate verification when the `scheme` is set to `HTTPS` [1] 

The username/password bits in the other parts is setting up HTTP Basic authentication primarily used here for role based(authorization) access to the API 

[1] https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#http-probes